design/ceph-csi-driver.md

```diff
@@ -54,6 +56,19 @@ If CSI drivers are already successfully deployed, Rook should help monitor the S
 
 ### Rook initiated CSI drivers deployment
 
-If Rook is instructed to deploy CSI drivers and other external controllers, Rook Operator should create all the driver and daemon deployments as well.
+If Rook is instructed to deploy CSI drivers and other external controllers, Rook Operator should create all the driver and daemon deployments as well. Since there are breaking changes in CSI 1.0 spec and Kubernetes 1.13, it makes sense to focus on newer CSI specs and only on latest Kubernetes releases (i.e. 1.13 and beyond).
+
+The changes to Rook include the following:
+
+- Create a new API object for CSI driver deployment. Such object should inform Rook Ceph Operator which driver (rbd or Ceph FS, or both) to start, in which namespace to run the CSI drivers, and which Ceph cluster to get configuration infomation to create Storage Class and Secret.
+- Operator starts CSI Drivers and Controllers according to the new API spec.
+
+In the near term (Kubernetes 1.13 and maybe 1.14 timeframe), we adopt the following approaches, allowing us to use latest CSI driver image and runtime configuration:
+- Use Golang template to materialize Kubernetes and Rook API objects.
```

Comment:
- Do you have an example of how the template will be used? 
- I'm confused about what it means that CSI declared 1.0 if their changes won't be backward compatible. 
- If K8s is updated after rook has already been started, will rook have to update the storage class, secrets, or other resources that will be compatible with the new version?